[#42]Member 도메인 클래스 수정

wooni97 opened the pull request.

wooni97 committed:
refactor: Member 도메인 클래스 수정

wooni97 committed:
refactor: Member 도메인 클래스 수정에 따른 MemberRepositoryImpl 메서드명 수정

f-lab-inno commented on src/main/java/dev/flab/studytogether/domain/member/repository/MemberRepositoryImpl.java:
```diff
@@ -28,16 +27,15 @@ public Member save(Member member) {
 
         Number key = jdbcInsert.executeAndReturnKey(new MapSqlParameterSource(parameters));
-        return new Member(key.intValue(), member.getId(), member.getPw(), member.getNickname());
+        return Member.createWithSequenceId((Integer) key, member.getId(), member.getPassword(), member.getNickname());
```
기존 intValue() 메소드 호출에서 형변환으로 변경하신 이유가 무엇일까요? `executeAndReturnKey`의 반환될 타입이 int 하위호환 타입인것임이 확실한 경우인것처럼 보이는데, Integer 타입으로 선언해도 되지 않을까요?

wooni97 replied:
이 부분은 제가 잘못 수정한 부분이네요! 다시 intValue() 메소드 호출로 변경했습니다.

f-lab-inno commented on src/main/java/dev/flab/studytogether/domain/member/entity/Member.java:
```diff
@@ -1,39 +1,34 @@
 
-    public Member(int seqID, String id, String pw, String nickname){
-        this.seqID = seqID;
+    private Member(int sequenceId, String id, String password, String nickname){
```
변수명이 조금 더 길어지더라도, 명시적인 표현 방법 아주 좋습니다! 👍

f-lab-inno commented on src/main/java/dev/flab/studytogether/domain/member/entity/Member.java:
```diff
@@ -1,39 +1,34 @@
-        return pw;
+    public static Member createWithoutSequenceId(String id, String password, String nickname) {
+        return new Member(id, password, nickname);
     }
-    public String getNickname(){
-        return nickname;
+    
+    public static Member createWithSequenceId(int sequenceId, String id, String password, String nickname) {
+        return new Member(sequenceId, id, password, nickname);
     }
```
객체 생성에 따른 생성자 이름 부여 차원의 정적 팩토리 메소드 패턴 사용 사례도 좋네요.

wooni97 committed:
refactor: 타입 캐스팅 수정

해당 커밋은 Member.createWithSequenceId 메서드 호출에서 `(Integer) key`를 `key.intValue()`로 수정합니다.
